daemon/cluster/executor/container/adapter.go

```diff
@@ -247,10 +247,6 @@ func (c *containerAdapter) create(ctx context.Context) error {
 		return errors.New("unable to get container from task spec")
 	}
 
-	if err := c.backend.SetContainerDependencyStore(cr.ID, c.dependencies); err != nil {
-		return err
-	}
```

Comment:
The implementation of the `SetContainerDependencyStore` method was deleted in this past commit as "swarm functionality": https://github.com/resin-os/balena/commit/82b451966319a5fe9a697dff5a28551eadb8b1e8#diff-304ae9de92630e351d339350b7c79a65L8
But this call was left behind and the unit tests were failing. As for the question, "how come balena was working with this call in the code", presumably the answer is that either this code was not being executed at run time, or the object implementing the the Backend interface is not the same Daemon struct used by the unit tests.